integration/spark/src/test/java/io/openlineage/spark/agent/lifecycle/StaticExecutionContextFactory.java

```diff
@@ -44,7 +44,7 @@ public StaticExecutionContextFactory(OpenLineageContext sparkContext) {
    * @throws InterruptedException
    */
   public static void waitForExecutionEnd() throws InterruptedException, TimeoutException {
```

Comment:
Can we instead of such method, create a spark listener which will notify us when the job would be done and we can proceed?